content/blog/2024/12/how-to-migrate-from-idf-bootloader-to-mcuboot-using-esp-self-reflasher/index.md

```diff
@@ -0,0 +1,270 @@
+---
+title: "How to migrate from IDF Bootloader to MCUboot using ESP Self-Reflasher"
+date: 2024-12-16
+showAuthor: false
+authors:
+  - "almir-okato"
+tags: ["Component", "ESP-IDF", "Bootloader", "MCUboot", "ESP Self Reflasher", "NuttX"]
+---
+
+Migration in Embedded Systems is always a significant subject and it may not only feature on an ongoing production decision, but also as a factor that can contribute to design choices at project time. Hardware or firmware wise, any change of that kind is risky, so it requires good planning, controled testing and careful execution.
+
+On the firmware area, there are a variety of reasons that may motivate choosing a flexible platform for a project or actually migrating from a production running RTOS (Real Time Operating System) to another one, they may range from technical to strategic purposes.
+
+Relatively, changing a RTOS may not be as straight forward as a firmware update. One of the concerns that needs to be addressed beforehand is the bootloader compatibility. Besides initializing the system, the bootloader may be responsible for handling safe and secure firmware updates, however usually it cannot update itself or it is not "updatable" by design or due device restriction features, e.g. flash bootloader region protection. Also in most of the scenarios, the device is not physically accessible, thus a direct serial flashing is not practical.
+
+This tutorial will demonstrate, how ESP Self-Reflasher component can be used to migrate an ESP32 embedded with an IDF-based application (OTA capable) + IDF bootloader to MCUboot compatible application + MCUboot bootloader. The ESP Self-Reflasher component is used on an middle step application for enabling the device to "reflash" its own firmware.
+
+---
+**Note**
+The MCUboot compatible application used in this guide is a NuttX RTOS application, but could be other like a Zephyr RTOS application, for instance.
+
+---
+
+The overall process described in this guide can be illustrated as:
+
+```mermaid
+flowchart LR
+    A[IDF bootloader + IDF application with OTA capabilities] -->|OTA process| B[Reflashing application with ESP Self-Reflasher component]
+    B -->|Reflashing process| C(MCUboot bootloader + NuttX application)
+```
+
+## ESP Self Reflasher component
+
+The [ESP Self-Reflasher](https://components.espressif.com/components/espressif/esp-self-reflasher/versions/1.0.0) is an ESP-IDF component that allows the device to reflash itself with new firmware. This is achieved through one of the following ways of operation: downloading the reflashing image or embedding the reflashing image. The component has examples for each that will be used on this guide.
```

Comment:
I would re-write this part to something similar to:
"..to be reflashed with new firmware."